src/Table/Table.js

```diff
@@ -74,14 +81,21 @@ const Table = React.forwardRef(({ headers, tableData, className, tableBodyClassN
                     }
 
                     if (richTable) {
-                        checkboxCell = <td className={tableCheckboxClasses}>{row.rowData[0]}</td>;
+                        checkboxCell = (
+                            <td
+                                className={tableCheckboxClasses}
+                                onChange={(e) => onCheckboxClick(e, index)}>
+                                {row.rowData[0]}
+                            </td>
+                        );
                         displayRows = row.rowData.splice(1, row.rowData.length);
                     }
 
                     return (
                         <tr
                             className={tableRowClasses}
                             {...rowProps}
+                            aria-selected={selectedRows[index]}
```

Comment:
If we know the first item in the row is a checkbox, can we just do
```js
aria-selected={row?.rowData?[0]?.props.checked}
```
Otherwise we'd be having duplicated selected logic here and the consumer, like Table.stories.js, and they could get out of sync.  Such as if they wanted to render a row pre-selected this component would not account for that.